src/onegov/form/parser/grammar.py

```diff
@@ -603,7 +603,7 @@ def marker_box(characters: str) -> ParserElement:
     #       If pricing wasn't optional this would be easy...
     label = Regex(
         # a sequence of words (that can't start with brackets)
-        rf'(?P<label>{text_without_re(characters + "()")}(?: ?{word_re})*?) *'
+        rf'(?P<label>{text_without_re(characters + "()")}(?: *{word_re})*?) *'
         # followed by optional pricing or discount followed by end of line
         r'(?=$|[(] *-?[0-9]+(?:\.[0-9]+)? *(?:%|[A-Za-z]{3}!?) *[)] *$)'
```

Comment:
```suggestion
        r'(?=$| *[(] *-?[0-9]+(?:\.[0-9]+)? *(?:%|[A-Za-z]{3}!?) *[)]|  )'
```
Let's try this instead. It will make one of the test cases fail where there's a label with a price and then something after the price, but that's fine, we'll just live with that behavior (i.e. it discards everything after the price information)

Reply:
Hang on the first part will need to change slightly as well, I will make a new comment with the full suggested change